tests/linearizability/model.go

```diff
@@ -209,5 +231,51 @@ func stepDelete(state EtcdState, request EtcdRequest, response EtcdResponse) (bo
 
 	state.Value = ""
 	state.LastRevision = response.Revision
+	state.FailedWrite = nil
+	return true, state
+}
+
+func stepTxn(state EtcdState, request EtcdRequest, response EtcdResponse) (bool, EtcdState) {
+	if response.Err != nil {
+		state.FailedWrite = &request
+		return true, state
+	}
+	// revision should never decrease
+	if response.Revision < state.LastRevision {
+		return false, state
+	}
```

Comment:
I recall that you added this check in the `step` function in another PR. So pls remove this now or after you merge the other PR.